gdcdatamodel/models/indexes.py

```diff
@@ -80,7 +80,20 @@ def get_secondary_key_indexes(cls):
         ) for key in secondary_keys
     )
 
-    return tuple(key_indexes) + tuple(lower_key_indexes)
+    rv = tuple(key_indexes) + tuple(lower_key_indexes)
+
+    if secondary_keys:
```

Comment:
Oops, this is wrong - it's a flatten 1-dimension array, not the original 2-dimension array.